enhancements/cloud-integration/aws/aws-pod-identity.md

```diff
@@ -0,0 +1,569 @@
+
+```go
+// AWSStatus containes the status of the credentials request in AWS.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type AWSProviderStatus struct {
+        metav1.TypeMeta `json:",inline"`
+        // User is the name of the User or Role created in AWS for these credentials.
+        User string `json:"user"`
+        // Policy is the name of the policy attached to the user in AWS.
+        Policy string `json:"policy"`
+}
+```
+
+#### `CredentialRequest` Example (Phase 2)
+
+A `CredentialsRequest` CR like this
+```yaml
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: openshift-image-registry
+  namespace: openshift-cloud-credential-operator
+spec:
+  storage:
+    serviceAccount:
+      name: image-registry-sa
+      namespace: openshift-image-registry
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - effect: Allow
+      action:
+      - "s3:*"
+      resource: "*"
+```
+
+Would result in the creation of an IAM role named
+`sjenning-abcde-openshift-image-registry` with `policy.json`
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:*"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+and a `trust.json`
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::111122223333:oidc-provider/s3-us-west-1.amazonaws.com/sjenning-abcde-oidc-provider"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "__doc_comment": "scope the role to the service account (optional)",
+        "StringEquals": {
+          "s3-us-west-1.amazonaws.com/sjenning-abcde-oidc-provider:sub": "system:serviceaccount:openshift-image-registry:image-registry-sa"
+        }
+      }
+    }
+  ]
+}
+
+```
+where `111122223333` is the ARN ID of the OIDC provider.
+
+The controller then creates the `ServiceAccount`
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: image-registry-sa
+  namespace: openshift-image-registry
+  annotations:
+    eks.amazonaws.com/role-arn: "arn:aws:iam::111122223333:role/sjenning-abcde-openshift-image-registry"
+```
+
+Any pod started with this `serviceAccountName` will have the token injected via
+the pod identity webhook and be able to assume the IAM role annotated in the
+`ServiceAccount`.
+
+### Risks and Mitigations
+
+Access to the new functionality is gated by two things: the platform being
+`aws` and, once phase 2 is complete, the user specifying a `spec.storage.serviceAccount` in the `CredentialsRequests` CR.
+
+## Design Details
```

Comment:
Looks like everything after this is template boilerplate.  Maybe a followup PR where you drop the sections?  Or explicitly say "We do not need a test plan for this enhancement" and so forth for each section?